Add support to resource aware api_root #156

humrochagf opened the pull request.

filipeximenes commented on tapioca/tapioca.py:
```diff
@@ -110,7 +110,7 @@ def _get_client_from_name(self, name):
         if name in resource_mapping:
             resource = resource_mapping[name]
-            api_root = self._api.get_api_root(self._api_params)
+            api_root = self._api.get_api_root(self._api_params, name)
```
Just for safety, in case we add more params to the method later, should we pass this as kwarg? `resource_name=name`

humrochagf replied:
That is a good idea, I will change it and update the PR

humrochagf replied:
Thinking a little about it the downside is that it obscures the public api because who calls get_api_root and pass it parameters is the client internals. The public interface with the customer will only display kwargs.

This pattern makes more sense when you are adding parameters to a context where you are the consumer, like at django class based views when you need to add variables to the context data to use them at the template.

But in this case, where you receive these variables to make the decision of which api_root to use, it's more helpful to know the parameters available.

humrochagf replied:
What do you think about it @filipeximenes? In any case, I just remembered that I need to update the docs, after the final decision I'll update it on this PR

filipeximenes replied:
Got it. Whenever it's a kwarg I prefer to pass it as kwarg in all situations I call it. I think it's more explicit. Those parameters are not mandatory, so in some way the [POLA](https://en.wikipedia.org/wiki/Principle_of_least_astonishment) does not apply (or it's less clear, because it's hard to guess what would people expect of it). 

I don't have much to back my point, it's just a stylish preference. That said, I leave it to you to decide what you prefer. As longs as it's a kwarg in the public api and that it's documented, I'm fine with it :slightly_smiling_face:

humrochagf replied:
Right, I'll pass as kwargs, add to the docs and add a header docstring to help anyone like me that first checks for the method header and function definition before going to the docs

humrochagf replied:
Right, I'll pass as kwargs, add to the docs and add a header docstring to help anyone like me that first checks for the method header and function definition before going to the docs

humrochagf commented:
Changed to use kwargs and updated the docs. The docstring at the method header I'll leave to another PR because it will require defining a pattern to apply to the other method headers :smile:
